conans/test/unittests/client/build/cpp_std_flags_test.py

```diff
@@ -1,195 +1,170 @@
 import unittest
 
-from conans.client.build.cppstd_flags import cppstd_default, cppstd_flag
+from conans.client.build.cppstd_flags import cppstd_default
+from conans.client.conf import get_default_settings_yml
+from conans.model.settings import Settings
+from conans.tools import cppstd_flag
 
 
-class CompilerFlagsTest(unittest.TestCase):
+def _make_cppstd_flag(compiler, compiler_version, cppstd):
+    settings = Settings.loads(get_default_settings_yml())
+    settings.compiler = compiler
+    settings.compiler.version = compiler_version
+    settings.compiler.cppstd = cppstd
+    return cppstd_flag(settings)
 
-    def test_gcc_cppstd_flags(self):
-        self.assertEqual(cppstd_flag("gcc", "4.2", "98"), "-std=c++98")
```

Comment:
Where are the tests that say that gcc 4.2 is c++98? 
Why removing this check?

Reply:
GCC 4.2 is not valid in defaults **settings.yml**, thus test may raise **ConanException**.
the initial version had `self.assertRaises` for cases like that, but @jgsogo suggested to remove it, as it wasn't testing a tool, but testing settings validation.
same for Clang 2.0 and others.
this might be kinda breaking, but as it was an internal tool, we can change it (unless we need to get valid flags for compilers not from **settings.yml** somewhere).

Reply:
Even if it is internally used, build-helpers were calling it. If users had added 4.2 to their settings.yml, are we sure it will keep working for them?

This is what I mean. In general, something called a "feature" should have really minimal changes to existing tests. The changes in these tests make very hard to know what is actually changing or not.

Reply:
so, do you think we should resurrect these tests and add GCC 4.2 and others to the test settings.yml?

Reply:
Yes, or mock the settings in ``_make_cppstd_flag()`` test helper, or something like that. Whatever is easier in test code.

Reply:
okay, changed to use mock, resurrected old tests